test/cookbooks/test/recipes/ruby.rb

```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 file '/tmp/pg_gem_test.rb' do
   content "#!/opt/chef/embedded/bin/ruby\nrequire 'pg'"
-  mode 0755
+  mode '755'
```

Comment:
Using strings here allow you to preserve any leading 0s. It's a rubocop-chef rule actually, which is what fixed this.